src/main/java/roomescape/service/MemberService.java

```diff
@@ -0,0 +1,39 @@
+package roomescape.service;
+
+import java.util.List;
+import java.util.Optional;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import roomescape.domain.Member;
+import roomescape.domain.repository.MemberRepository;
+import roomescape.dto.request.MemberRequest;
+import roomescape.dto.response.MemberResponse;
+import roomescape.exception.DuplicatedEmailException;
+
+@Service
+@Transactional
+public class MemberService {
+
+    private final MemberRepository memberRepository;
+
+    public MemberService(MemberRepository memberRepository) {
+        this.memberRepository = memberRepository;
+    }
+
+    @Transactional(readOnly = true)
+    public List<MemberResponse> findAllMembers() {
+        return memberRepository.findAll()
+                .stream()
+                .map(MemberResponse::from)
+                .toList();
+    }
+
+    public MemberResponse createMember(MemberRequest memberRequest) {
+        Optional<Member> optionalMember = memberRepository.findByEmail(memberRequest.email());
+        if (optionalMember.isPresent()) {
+            throw new DuplicatedEmailException();
+        }
+        Member savedMember = memberRepository.save(memberRequest.toMember());
+        return MemberResponse.from(savedMember);
+    }
```

Comment:
아래 처럼 수정하거나 exists를 활용할 수도 있을 것 같아요!

```suggestion
    public MemberResponse createMember(MemberRequest memberRequest) {
        memberRepository.findByEmail(memberRequest.email())
                .orElseThrow(DuplicatedEmailException::new);

        Member savedMember = memberRepository.save(memberRequest.toMember());
        return MemberResponse.from(savedMember);
    }
```

Reply:
제안 감사드립니다! 
exists를 활용하는 방법으로 리펙터링 진행하였습니다!